packages/flutter_tools/bin/podhelper.rb

```diff
@@ -288,7 +293,7 @@ def flutter_parse_plugins_file(file, platform)
 
   # dependencies_hash.dig('plugins', 'ios') not available until Ruby 2.3
   return [] unless dependencies_hash.has_key?('plugins')
-  return [] unless dependencies_hash['plugins'].has_key?('ios')
```

Comment:
How has this been working?

Reply:
`path_provider` currently doesn't share implementations though, so the macOS implementation package is macOS-only, and it works. 🤔